src/libs/ajax/data-table-providers/WdsDataTableProvider.ts

```diff
@@ -82,33 +83,59 @@ const getRelationParts = (val: unknown): string[] => {
   return []
 }
 
-// Extract wds URL from Leo response. exported for testing
-export const getWdsUrl = apps => {
-  // look explicitly for an app named 'wds-${app.workspaceId}'. If found, use it, even if it isn't running
-  // this handles the case where the user has explicitly shut down the app
+// Invokes logic to determine a URL for WDS
+// If WDS is not running, a URL will not be present -- in some cases, this function may invoke
+// a new call to Leo to instantiate a WDS being available, thus having a valid URL
+export const resolveWdsUrl = (apps, workspaceId) => {
+  // WDS looks for Kubernetes deployment statuses (such as RUNNING or PROVISIONING), expressed by Leo
+  // See here for specific enumerations -- https://github.com/DataBiosphere/leonardo/blob/develop/core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/kubernetesModels.scala
+  // look explicitly for a RUNNING app named 'wds-${app.workspaceId}' -- if WDS is healthy and running, there should only be one app RUNNNING
   const namedApp = apps.filter(app => app.appType === 'CROMWELL' && app.appName === `wds-${app.workspaceId}` && app.status === 'RUNNING')
   if (namedApp.length === 1) {
     return namedApp[0].proxyUrls.wds
   }
-  // if we didn't find the expected app 'cbas-wds-default', go hunting:
-  const candidates = apps.filter(app => app.appType === 'CROMWELL' && app.status === 'RUNNING')
+
+  // if we didn't find the expected app 'wds-${app.workspaceId}' running...
+  const candidates = apps.filter(app => app.appType === 'CROMWELL' && app.appName === `wds-${app.workspaceId}`)
+  // ...nothing has launched yet, bring WDS to life!
   if (candidates.length === 0) {
-    // no app deployed yet
+    const typedWithErrorReporting : any = withErrorReporting
+    const createLeoAppWithErrorHandling = typedWithErrorReporting('An error occurred when creating your data tables. Please reach out to support@terra.bio', async () => {
```

Comment:
Flag for Slack conversation w/ @msilva-broad

Reply:
glad you got this working, but I think we can do better then type any.... withErrorReporting is part of a family of curried functions that are challenging to have good types on... but I created some helpers (safeCurry) to help... let me poke around and see what a richer solution might look like for this.  I will post results.

Reply:
I could just type it as `Function`?